cirq/contrib/acquaintance/permutation.py

```diff
@@ -220,9 +220,11 @@ def update_mapping(mapping: Dict[ops.Qid, LogicalIndex],
         operations: The operations to update according to.
     """
     for op in ops.flatten_op_tree(operations):
-        if (isinstance(op, ops.GateOperation) and
-            isinstance(op.gate, PermutationGate)):
-            op.gate.update_mapping(mapping, op.qubits)
+        # Ignoring type warning about passing an abstract type into Type[...].
+        gate = ops.op_gate_of_type(op, PermutationGate)  # type: ignore
+        if gate is not None:
+            # Ignoring type warning about op.qubits not being a tuple.
+            gate.update_mapping(mapping, op.qubits)  # type: ignore
```

Comment:
Why does mypy think `op.qubits` has type `Union[Tuple[Qid, ...], FrozenSet[Qid]]`? Shouldn't it be a tuple?

Reply:
I honestly don't know. My guess is that someone in the code base a qubits method returns a FrozenSet[Qid] and mypy is inferring the return type from that, but I wasn't able to find where.

Reply:
It seems like a silly thing to spend too much time on, but this seems like a case where mypy might actually be catching an bug, so maybe we shouldn't override it so quickly.